DoubleBufferFloat.cpp

```diff
@@ -76,31 +62,74 @@ size_t DoubleBufferFloat::getData(float ** data, uint32_t * timestamp, bool swap
 	_isGetting = false;
 }
 
+bool DoubleBufferFloat::swap()
+{
+	// ToDo: add a mutex to handle multithreaded CPUs
+	
+	if (_outputBuffer != nullptr)
+	{
+		_outputBuffer->clear();
+		if (_inputBuffer == _buffer1) {
+#ifdef DEBUG
+			Serial.println("_inputBuffer == &_buffer2");
+#endif // DEBUG
+			_inputBuffer = _buffer2;
+			_outputBuffer = _buffer1;
+		}
+		else {
+#ifdef DEBUG
+			Serial.println("_inputBuffer != &_buffer1");
+#endif // DEBUG
+			_inputBuffer = _buffer1;
+			_outputBuffer = _buffer2;
+		}
+		return true;
+	}
+	return false;
+}
+
 //void DoubleBufferFloat::setAutoResize(bool b) {
 //	_buffer1->autoResize = b;
 //	_buffer2->autoResize = b;
 //}
 
+size_t DoubleBufferFloat::size(BufferSelector b)
+{
+	if (_getBuffer(b) == nullptr) return 0;
+	else return _getBuffer(b)->size();
+}
+
+
 size_t DoubleBufferFloat::inSize() {
-	if (_inputBuffer != nullptr) {
+	if (_inputBuffer != nullptr) return 0;
+	else {
 		_inputBuffer->size();
 	}
 }
 
 size_t DoubleBufferFloat::outSize() {
-	if (_outputBuffer != nullptr) {
+	if (_outputBuffer != nullptr) return 0;
+	else {
 		_outputBuffer->size();
 	}
 }
 
+size_t DoubleBufferFloat::capacity(BufferSelector b)
+{
+	if (_getBuffer(b) == nullptr) return 0;
+	else return _getBuffer(b)->capacity();
+}
+
 size_t DoubleBufferFloat::inCapacity() {
-	if (_inputBuffer != nullptr) {
+	if (_inputBuffer != nullptr) return 0;
+	else {
```

Comment:
Doesn't this translate to nullptr->capacity()?
That will definitely crash if someone calls inCapacity(). Is this correct?